backend/django/core/views/api_admin.py

```diff
@@ -323,3 +325,44 @@ def heat_map_data(request, project_pk):
         coders.append({"name": str(p.profile), "pk": p.profile.pk})
 
     return Response({"data": heatmap_data, "labels": labels, "coders": coders})
+
+
+@api_view(["GET"])
+@permission_classes((IsAdminOrCreator,))
+def get_project_status(request, project_pk):
+    """This returns the labeled data.
+
+    Args:
+        request: The GET request
+        project_pk: Primary key of the project
+    Returns:
+        data: a list of data information
+    """
+
+    project = Project.objects.get(pk=project_pk)
+
+    total_data_objs = project.data_set.all()
+
+    final_data_objs = list(
+        DataLabel.objects.filter(data__project=project, data__irr_ind=False)
+    )
+
+    stuff_in_queue = DataQueue.objects.filter(queue__project=project)
```

Comment:
One final thing for this (promise). The way queues work in SMART is pretty complicated. Since SMART is based on this idea that we want to label some data before others, queues are created for assignments to be pulled from.

There are three types of queues:
- admin: this is the one data are put in when they need an admin to look at them
- IRR: this is where data which is IRR and should be assigned first goes
- normal: this is where data which is not IRR and should be assigned first goes

So basically, data being in a queue doesn't mean it's been assigned yet, so we don't want to exclude all queues here. What we do want to exclude is everything in the admin queue (since that's at least been skipped or something), and all assigned data. Also anything in the IRRLog since that indicates someone has labeled/skipped that data.

All this to say I think the following change is needed.

```
    stuff_in_irrlog = IRRLog.objects.filter(data__project=project).values_list(
        "data__pk", flat=True
    )
    stuff_in_queue = DataQueue.objects.filter(queue__project=project, queue__type="admin").values_list(
        "data__pk", flat=True
    )
    recycle_ids = RecycleBin.objects.filter(data__project=project).values_list(
        "data__pk", flat=True
    )
    assigned_ids = AssignedData.objects.filter(data__project=project).values_list(
        "data__pk", flat=True
    )
    unlabeled_data_objs = list(
        project.data_set.filter(datalabel__isnull=True)
        .exclude(id__in=stuff_in_queue)
        .exclude(id__in=stuff_in_irrlog)
        .exclude(id__in=recycle_ids)
        .exclude(id__in=assigned_ids)
    )
```

Reply:
thanks for this! just added it

Reply:
Ok! I think this looks good